src/panels/RetinaCapturePanel.ts

```diff
@@ -0,0 +1,169 @@
+import { Uri, window, workspace } from "vscode";
+import { BasePanel, PanelDataProvider } from "./BasePanel";
+import { InitialState, ToVsCodeMsgDef } from "../webview-contract/webviewDefinitions/retinaCapture";
+import { MessageHandler } from "../webview-contract/messaging";
+import { TelemetryDefinition } from "../webview-contract/webviewTypes";
+import { withOptionalTempFile } from "../commands/utils/tempfile";
+import { KubectlVersion, invokeKubectlCommand } from "../commands/utils/kubectl";
+import * as k8s from "vscode-kubernetes-tools-api";
+import { failed } from "../commands/utils/errorable";
+import * as vscode from "vscode";
+import { longRunning } from "../commands/utils/host";
+import { relative } from "path";
+import { platform } from "os";
+import open from 'open';
+
+
+export class RetinaCapturePanel extends BasePanel<"retinaCapture"> {
+    constructor(extensionUri: Uri) {
+        super(extensionUri, "retinaCapture", {
+            startCaptureResponse: "",
+            getAllNodesResponse: [],
+        });
+    }
+}
+
+export class RetinaCaptureProvider implements PanelDataProvider<"retinaCapture"> {
+    constructor(
+        readonly kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+        readonly kubeConfigFilePath: string,
+        readonly kubectlVersion: KubectlVersion,
+        readonly clusterName: string,
+        readonly retinaOutput: string,
+        readonly allNodeOutput: string[],
+        readonly captureFolderName: string
+    ) { }
+
+    getTitle(): string {
+        return `Retina Distributed Capture on ${this.clusterName}`;
+    }
+
+    getTelemetryDefinition(): TelemetryDefinition<"retinaCapture"> {
+        return {
+            getAllNodes: false,
+            runRetinaCapture: true,
+            retinaCaptureResult: true,
+        }
+    }
+
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            retinaOutput: [this.retinaOutput],
+            allNodes: this.allNodeOutput,
+            selectedNode: "",
+            captureFolderName: this.captureFolderName,
+        };
+    }
+
+    getMessageHandler(): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            runRetinaCapture: (node: string) => this.handleRunRetinaCapture(node),
+            retinaCaptureResult: () => { },
+            getAllNodes: () => { }
+        };
+    }
+
+    private async handleRunRetinaCapture(node: string) {
+        const localCaptureUri = await window.showSaveDialog({
+            defaultUri: Uri.file(this.captureFolderName),
+            saveLabel: "Download",
+            title: "Download Retina File",
+        });
+
+        if (!localCaptureUri) {
+            return;
+        }
+
+        const localCpPath = getLocalKubectlCpPath(localCaptureUri);
+
+        const createPodYaml = `
+apiVersion: v1
```

Comment:
I think may be for future we can extract this to template file and use it for readability. leaving here as comment just to remember this and if you agree @Tatsinnit

Reply:
Thanks for noticing this, Yea, that's the plan, there is good about of clean is now in order for this repo, so bridging this gap for n/w user exp and align that, doing iteratively more close related work PR I guess.